### PR TITLE
Adds an advanced Workbox recipe for a custom postMessagePlugin

### PR DESCRIPTION
**Pull request by jeffposnick**
**R:** @gauntface 

**Fixes:** https://github.com/GoogleChrome/workbox/issues/1326

**CC:** @petele

### CODE REVIEW

File: src/content/en/tools/workbox/guides/advanced-recipes.md [hunk 1]
```diff
@@ -173,3 +173,44 @@ worbox.routing.registerRoute(new RegExp('/images/'), ({event}) => {
     .catch(() => caches.match(FALLBACK_IMAGE_URL));
 });
 ```
+
+## Use postMessage() to notify of cache updates
+
+Not every browser which supports service workers also supports the
```
Comment (gauntface) [inline src/content/en/tools/workbox/guides/advanced-recipes.md]: Optional: You could drop this first sentence.

File: src/content/en/tools/workbox/guides/advanced-recipes.md [hunk 2]
```diff
@@ -173,3 +173,44 @@ worbox.routing.registerRoute(new RegExp('/images/'), ({event}) => {
...
+[Broadcast Channel API](https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API).
+On browsers which lack [Broadcast Channel API support](https://caniuse.com/#search=broadcast), the
+`workbox-broadcast-cache-update` plugin will not send any messages letting clients know that a
+cached response was updated.
+
+As an alternative to using `workbox-broadcast-cache-update`, you can instead "roll your own" plugin
+which listens for the same cache update events, and uses the more widely supported [`postMessage()`
+API](https://developer.mozilla.org/en-US/docs/Web/API/DedicatedWorkerGlobalScope/postMessage) for
+sending out notifications. In this custom plugin, you have control over what criteria are
```
Comment (gauntface) [inline src/content/en/tools/workbox/guides/advanced-recipes.md]: Nit: Swap 'notifications' for 'updates'.

File: src/content/en/tools/workbox/guides/advanced-recipes.md [hunk 3]
```diff
@@ -173,3 +173,44 @@ worbox.routing.registerRoute(new RegExp('/images/'), ({event}) => {
...
+```javascript
+const postMessagePlugin = {
+  cacheDidUpdate: async ({cacheName, url, oldResponse, newResponse}) => {
+    // Use whatever logic you want to determine whether the responses differ.
+    if (oldResponse && (oldResponse.headers.get('etag') !== newResponse.headers.get('etag'))) {
+      const clients = await self.clients.matchAll();
+      for (const client of clients) {
+        // Use whatever message body makes the most sense.
+        // Note that `Response` objects can't be serialized.
+        client.postMessage({url, cacheName});
```
Comment (josephliccini) [inline src/content/en/tools/workbox/guides/advanced-recipes.md]: might want this in the Flux format to match what you do for `BroadcastChannel`.  Up to you though just throwing it out there :) thats what we did in our plugin

Comment (jeffposnick) [reply src/content/en/tools/workbox/guides/advanced-recipes.md]: I'd like to just leave it open for folks to add in whatever data they need.

**Pull request merged**